cli/lsp/documents.rs

```diff
@@ -781,48 +865,49 @@ impl DocumentsInner {
     }
   }
 
-  fn contains_specifier(&mut self, specifier: &ModuleSpecifier) -> bool {
-    let specifier = self
-      .resolve_specifier(specifier)
-      .unwrap_or_else(|| specifier.clone());
-    if !self.is_valid(&specifier) {
-      self.add(specifier.clone());
-    }
-    self.docs.contains_key(&specifier)
+  fn contains_specifier(&self, specifier: &ModuleSpecifier) -> bool {
+    self.get(specifier).is_some()
```

Comment:
you are convinced that cloning a document is lightweight?  This will clone any in memory documents just to find out if they are present.